File: 1669.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        start = list1
        b = b - a
        for i in range(a - 1):
            start = start.next
        end = start
        for i in range(b + 2):
            end = end.next
        start.next = list2
        while start.next:
            start = start.next
        start.next = end
        return list1
            

        return list1

        

File: 1669.cpp
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode* prev = list1;
        for (int i = 0; i < a - 1; ++i) {
            prev = prev->next;
        }

        ListNode* curr = prev;
        for (int i = 0; i <= b - a + 1; ++i) {
            curr = curr->next;
        }

        ListNode* tail2 = list2;
        while (tail2->next) {
            tail2 = tail2->next;
        }
        tail2->next = curr;
        prev->next = list2;
        return list1;
    }
};
